tests/wpt/mozilla/tests/mozilla/img-find-areas.html

```diff
@@ -0,0 +1,46 @@
+<!doctype html>
+    <meta charset="utf-8">
+    <title>Test for issue #15884</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <body>
+        <map name="img_map" id="img_map">
+            <area shape="rect" coords="0,0,100,100"/>
+        </map>
+        <img usemap="#img_map" src="2x2.png" id="img"/>
+
+        <img usemap="#img_no_map" src="2x2.png" id="img_no"/>
+        <script>
+            var test = async_test("ImageShouldFindTheMapEvenWhenItIsNotItsSibiling");
+            var img = document.getElementById('img');
+            var img_clicked = false;
+            img.onclick = function(e) {
+                img_clicked = true;
+                return true;
+            };
+            document.addEventListener("DOMContentLoaded", function() {
+                test.step(function() {
+                    img.click()
+                });
+            });
+            test.step_timeout(function() {
+                assert_true(img_clicked, "Image click expected"); test.done();
+            }, 500);
+
+            var img_no = document.getElementById('img_no');
+            var img_no_map_clicked = false;
+            img_no.onclick = function(e) {
```

Comment:
`img_no.onclick = test2.unreached_func("click event should not be fired for image without map");`